src/components/sbb-toggle-check/sbb-toggle-check.tsx

```diff
@@ -0,0 +1,108 @@
+import { Component, Prop, h, Event, EventEmitter } from '@stencil/core';
+import { InterfaceCheckToggleAttributes } from './sbb-link.custom';
+
+@Component({
+  shadow: true,
+  styleUrl: 'sbb-toggle-check.scss',
+  tag: 'sbb-toggle-check',
+})
+export class SbbToggleCheck {
+  private _checkbox: HTMLInputElement;
+
+  /** Whether the toggle-check is checked. */
+  @Prop({ mutable: true }) public checked = false;
+
+  /** name of the toggle-check */
+  @Prop() public name?: string;
+
+  /** id of the toggle-check */
+  @Prop() public toggleId: string;
+
+  /** Id which is sent in the change event payload */
+  @Prop() public eventId?: string;
+
+  /** the svg name for the true state - default -> 'tick-small' */
+  @Prop() public icon = 'tick-small';
+
+  /** the disabled prop for the disabled state */
+  @Prop() public disabled!: boolean;
+
+  /** The label position relative to the toggle. Defaults to 'after' */
+  @Prop() public labelPosition?: InterfaceCheckToggleAttributes['labelPosition'] = 'after';
+
+  /** the aria-label prop for the hidden input */
+  @Prop() public accessibilityLabel?: string;
+
+  /** the aria-labelledby prop for the hidden input */
+  @Prop() public accessibilityLabelledby?: string;
+
+  /** the aria-describedby prop for the hidden input */
+  @Prop() public accessibilityDescribedBy?: string;
+
+  /** event for emiting whenever selection is changed */
+  @Event() public changed: EventEmitter<boolean>;
+
+  /** set checked to the state of the input-checkbox */
+  private _toggle(): void {
```

Comment:
To elaborate a bit on the state management of the checkbox and checked, because it is more complex than it might seem.
For the changed event (naming of events is still an open topic, usually we prefix it) you can essentially just watch the checked property:
```ts
@Watch('checked')
  checkedChanged(isChecked: boolean) {
    this.sbbChange.emit({
      checked: isChecked,
      value: this.value,
    });
  }
```

On the input element, you can use the `click` event to listen to changes and update `checked` accordingly.
Create a unit or e2e test to test the scenarios of changing checked programmatically, via click and via keyboard.

We might also have to call `.stopPropagation()` on the input event, as the label solution might cause an additional click event to be triggered. (See https://github.com/sbb-design-systems/sbb-angular/blob/main/src/angular/checkbox/checkbox.ts#L288-L295 )

We can discuss this further via Teams if you want, as I'm running out of time for today.

Reply:
@kyubisation I understand this is merged already, but Osman just pointed me here. Any reason you would maintain/duplicate state inside the component (checked) instead of using the passed value/checked prop directly? I'd generally suggest to emit the event directly from the onChange-Event from the input field to keep the comp itself stateless.

Reply:
It is not possible to avoid state because Stencil does not support getter/setter for properties. For a custom checkbox it is essential to have a public checked property which should mirror the inner checked state.

Reply:
Props are not reactive?